### PR TITLE
fix warning with objects in session

### PR DESCRIPTION
**Pull request by btry**
A proposal to fix warnings in debug mode when QuerySubQuery objects are added in $_SESSION['glpicondition']

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #6546

### CODE REVIEW

File: inc/querysubquery.class.php [hunk 1]
```diff
@@ -74,4 +74,8 @@ public function getQuery() {
       }
       return $sql;
    }
+
+   public function __toString() {
```
Comment (trasher) [inline inc/querysubquery.class.php]: I guess it should be implemented in:
- AbstractQuery
- QueryParam
- QueryExpression

File: inc/html.class.php [hunk 2]
```diff
@@ -3925,7 +3925,7 @@ static function printCleanArray($tab, $pad = 0, $jsexpand = false) {
                   }
                } else {
                   if (is_object($val)) {
-                     print_r($val);
+                     echo (string) $val; // It is expected that the object implements __toString()
```
Comment (trasher) [inline inc/html.class.php]: Thinking about it, maybe could we add an interface for Queries objects, and add a test on the interface implementation  before the `is_object` (and keep `print_r` for this case).

Comment (btry) [reply inc/html.class.php]: I agree

As the involved classes are used as substring of a final SQL query string, I suggest one of the following names
- QueryFragmentInterface
- QueryStringInterface

Comment (trasher) [reply inc/html.class.php]: Or maybe just test for the `__toString` method? 4ve never tried, dunno if that works :)

Comment (btry) [reply inc/html.class.php]: if will work
```php
Interactive mode enabled

php > class foo { }
php > class bar { public function __toString() { return ''; }}
php > var_dump(method_exists('foo', '__toString'));
php shell code:1:
bool(false)
php > var_dump(method_exists('bar', '__toString'));
php shell code:1:
bool(true)
```

Comment (btry) [reply inc/html.class.php]: this is simpler, and is future proof IMHO.

Comment (trasher) [reply inc/html.class.php]: :+1:

**Pull request merged**